keras_cv/layers/preprocessing/rgb_shift.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np 
+import tensorflow as tf
+from tensorflow.keras import layers, backend
+
+
+class RGBShift(layers.Layer):
+    """RGBShift class randomly shift values for each channel of the input RGB image. The expected images 
+    should be [0-255] pixel ranges.
+
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Args:
+        factor: A scalar or tuple or list of two upper and lower bound number. If factor is a single value, 
+                the range will be (-factor, factor). The factor value can be float or integer; for float the 
+                valid limits are (-1.0, 1.0) and for integer the valid limits are (-255, 255).
+        seed: Integer. Used to create a random seed. Default: None.
+
+    Call arguments: call method for the RGBShift layer.
+        Args:
+            images: Tensor representing images of shape
+                [batch_size, width, height, channels], with dtype tf.float32 / tf.uint8, or,
+                [width, height, channels], with dtype tf.float32 / tf.uint8
+        Returns:
+            images: augmented images, same shape as input.
+   
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    rgbshift = keras_cv.layers.RGBShift(factor=(-2, 2))
+    augmented_images = rgbshift(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        factor,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.factor = self._set_shift_limit(factor)
+        self.seed = seed
+
+    def _set_shift_limit(self, factor):
+        if isinstance(factor, (tuple, list)):
+            if len(factor) != 2: 
+                raise ValueError(f'The factor should be scalar, tuple or list of two upper and lower \
+                    bound number. Got {factor}')
+            return self._check_factor_range(sorted(factor))
+        elif isinstance(factor, (int, float)):
+            factor = abs(factor)
+            return self._check_factor_range([-factor, factor])
+        else:
+            raise ValueError(f'The factor should be scalar, tuple or list of two upper and lower bound\
```

Comment:
Also fix this following the example in my comment above

Reply:
Done.